NPE when calling com.zsmartsystems.zigbee.ZigBeeNode.updateBindingTable()

We called `com.zsmartsystems.zigbee.ZigBeeNode.updateBindingTable()` on a Bitron Video Repeater AV2010/36. There we get a `ManagementBindResponse` with status `NOT_PERMITTED` and with empty binding table data. As there is no handling for that situation the method runs in a NPE on line `if (response.getStartIndex() == index) {`.

```
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(Unknown Source)
	at java.util.concurrent.FutureTask.get(Unknown Source)
	...
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException
	at com.zsmartsystems.zigbee.ZigBeeNode$1.call(ZigBeeNode.java:411)
	at com.zsmartsystems.zigbee.ZigBeeNode$1.call(ZigBeeNode.java:387)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	... 1 more
```

I think `NOT_PERMITTED` should be handled similar to `NOT_SUPPORTED`.

I attached a tiny log file containing the request and the response.
[logs.txt](https://github.com/zsmartsystems/com.zsmartsystems.zigbee/files/4031013/logs.txt)

--- a/com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/ZigBeeNode.java
+++ b/com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/ZigBeeNode.java
@@ -407,6 +407,12 @@ public class ZigBeeNode implements ZigBeeCommandListener {
                     if (response.getStatus() == ZdoStatus.NOT_SUPPORTED) {
                         return ZigBeeStatus.UNSUPPORTED;
                     }
+                    if (response.getStatus() == ZdoStatus.NOT_PERMITTED) {
+                        return ZigBeeStatus.INVALID_STATE;
+                    }
+                    if (response.getStatus() != ZdoStatus.SUCCESS) {
+                        return ZigBeeStatus.FAILURE;
+                    }
 
                     if (response.getStartIndex() == index) {
                         tableSize = response.getBindingTableEntries();

--- a/com.zsmartsystems.zigbee/src/test/java/com/zsmartsystems/zigbee/ZigBeeNodeTest.java
+++ b/com.zsmartsystems.zigbee/src/test/java/com/zsmartsystems/zigbee/ZigBeeNodeTest.java
@@ -602,8 +602,23 @@ public class ZigBeeNodeTest {
         nodeResponse.setSourceAddress(new ZigBeeEndpointAddress(123));
         nodeResponse.setDestinationAddress(new ZigBeeEndpointAddress(0));
         responses.put(ZdoCommandType.MANAGEMENT_BIND_REQUEST.getClusterId(), nodeResponse);
-
         Future<ZigBeeStatus> future = node.updateBindingTable();
         assertEquals(ZigBeeStatus.UNSUPPORTED, future.get());
+
+        nodeResponse = new ManagementBindResponse();
+        nodeResponse.setStatus(ZdoStatus.NOT_PERMITTED);
+        nodeResponse.setSourceAddress(new ZigBeeEndpointAddress(123));
+        nodeResponse.setDestinationAddress(new ZigBeeEndpointAddress(0));
+        responses.put(ZdoCommandType.MANAGEMENT_BIND_REQUEST.getClusterId(), nodeResponse);
+        future = node.updateBindingTable();
+        assertEquals(ZigBeeStatus.INVALID_STATE, future.get());
+
+        nodeResponse = new ManagementBindResponse();
+        nodeResponse.setStatus(ZdoStatus.NO_DESCRIPTOR); // Random error
+        nodeResponse.setSourceAddress(new ZigBeeEndpointAddress(123));
+        nodeResponse.setDestinationAddress(new ZigBeeEndpointAddress(0));
+        responses.put(ZdoCommandType.MANAGEMENT_BIND_REQUEST.getClusterId(), nodeResponse);
+        future = node.updateBindingTable();
+        assertEquals(ZigBeeStatus.FAILURE, future.get());
     }
 }

